src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs

```diff
@@ -301,7 +324,7 @@ public IAsyncResult BeginConnect(IPAddress address, int port, AsyncCallback requ
         {
             if (NetEventSource.IsEnabled) NetEventSource.Enter(this, address);
 
-            IAsyncResult result = Client.BeginConnect(address, port, requestCallback, state);
+            IAsyncResult result = NonDisposedClient.BeginConnect(address, port, requestCallback, state);
```

Comment:
What is this Client => NonDisposedClient fixing?  Couldn't it still become disposed immediately after we check for that in the property such that we're still operating on a disposed socket?

Reply:
Yes. This is ensuring we see ObjectDisposedException, instead of NullReferenceException (because the Client was set to null in the Dispose method).

Reply:
Why wouldn't we instead just not null out the Socket?  What do we gain by doing so?

Reply:
Some methods are implemented so they don't throw for a Disposed TcpClient by checking if the Socket is null:
```
public int Available => _clientSocket?.Available ?? 0;
public bool Connected => _clientSocket?.Connected ?? false;
...
```

Reply:
Sure, but why couldn't they just check cleanedup instead?